crud/src/main/java/com/redhat/lightblue/mediator/DefaultMetadataResolver.java

```diff
@@ -180,25 +185,54 @@ private void addFieldRoles(Set<String> roles, FieldTreeNode node) {
         }
     }
 
-    private EntityMetadata getMetadata(String entityName,
-                                       String version) {
-        EntityMetadata emd = metadataMap.get(entityName);
-        if (emd != null) {
-            if (!emd.getVersion().getValue().equals(version)) {
-                throw Error.get(CrudConstants.ERR_METADATA_APPEARS_TWICE, entityName + ":" + version
-                        + " and " + emd.getVersion().getValue());
-            }
-        } else {
-            LOGGER.debug("Retrieving entity metadata {}:{}", entityName, version);
-            emd = md.getEntityMetadata(entityName, version);
-            if (emd == null || emd.getEntitySchema() == null) {
-                throw Error.get(CrudConstants.ERR_UNKNOWN_ENTITY, entityName + ":" + version);
+    private void initMetadataMap(EntityMetadata emd, String entityName, String rootRequestVersion) {
+        // store root entity metadata
+        metadataMap.put(entityName, emd);
+        Set<String> deferred = new HashSet<>();
+        FieldCursor cursor = emd.getFieldCursor();
+        while (cursor.next()) {
+            FieldTreeNode node = cursor.getCurrentNode();
+            validateNode(emd, entityName, rootRequestVersion, deferred, node);
```

Comment:
I thought the cursor wouldn't traverse down into other objects?  Or did I misunderstand that?  Will they traverse everything but referenced objects?